main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -228,10 +229,7 @@ object Typer {
 
           run(initialSubst) match {
             case Ok((subst0, (partialTconstrs, partialType))) =>
-              ///
-              /// Propagate type variables *names* in the substitution.
-              ///
-              val subst = subst0.propagate
+              val subst = processTypeVariableNames(subst0, tparams0)
```

Comment:
I think we can call this `pivot` or `rotate` because it is really an operation that preserves the solution to the constraint system but rotates it along another axis.